JAVA-2054: Add now_in_seconds to protocol v5 query messages

olim7t opened the pull request.

olim7t committed:
JAVA-2054: Add now_in_seconds to protocol v5 query messages

olim7t commented on bom/pom.xml:
```diff
@@ -61,7 +61,7 @@
         <groupId>com.datastax.oss</groupId>
         <artifactId>native-protocol</artifactId>
-        <version>1.4.9</version>
+        <version>1.4.10-SNAPSHOT</version>
```
This requires datastax/native-protocol#34.

olim7t commented on core/src/main/java/com/datastax/oss/driver/api/core/cql/Statement.java:
```diff
@@ -348,6 +348,20 @@ default SelfT setRoutingKey(@NonNull ByteBuffer... newRoutingKeyComponents) {
+  @SuppressWarnings("unchecked")
+  default SelfT setNowInSeconds(int nowInSeconds) {
+    return (SelfT) this;
+  }
```
Since this feature is mainly intended for tests, I think default implementations are a good compromise to avoid breaking the API.

tomekl007 commented on core/src/main/java/com/datastax/oss/driver/internal/core/cql/Conversions.java:
```diff
@@ -142,6 +142,12 @@ public static Message toMessage(
     ProtocolVersionRegistry protocolVersionRegistry = context.getProtocolVersionRegistry();
     CqlIdentifier keyspace = statement.getKeyspace();
+    int nowInSeconds = statement.getNowInSeconds();
+    if (nowInSeconds != Integer.MIN_VALUE
```
I would extract `Integer.MIN_VALUE` and name it as a constant, for example:
`NOW_IN_SECONDS_UNSET` because it is used in a lot of places

tomekl007 commented on integration-tests/src/test/java/com/datastax/oss/driver/core/cql/NowInSecondsIT.java:
```diff
@@ -0,0 +1,122 @@
+
+  private static SessionRule<CqlSession> buildSessionRule() {
+    // Reminder to revisit the test when V5 comes out of beta: remove the custom config loader and
+    // inline this method.
```
Do you mean when the `V5` will be set as default? (so we will not need to do:
`.withString(DefaultDriverOption.PROTOCOL_VERSION, "V5")`)

olim7t replied:
I mean when `v5.isBeta()` becomes false.

olim7t committed:
Extract Statement.NO_DEFAULT_TIMESTAMP constant

adutra commented on core/src/main/java/com/datastax/oss/driver/api/core/cql/Statement.java:
```diff
@@ -67,6 +68,20 @@
+  /**
+   * A special value for {@link #getQueryTimestamp()} that means "no value".
+   *
+   * <p>It is equals to {@link Long#MIN_VALUE}.
```
Nit: "it is equal to". Same for the field below.

olim7t committed:
Fix dependency plugin configuration

Only redirect to a file for the `generate-dependency-list` execution, so
that command line invocations (e.g. `mvn dependency:tree`) still use the
standard output as expected.

olim7t committed:
Fix install_snapshots.sh

Since the introduction of the BOM (8f8e2e7), the native protocol version
is no longer expressed as a property in the parent POM.

Use the dependency plugin instead, this is more robust than a simple
grep.
